messaging/src/main/java/org/axonframework/messaging/configuration/CommandModelComponent.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.messaging.configuration;
+
+import jakarta.annotation.Nonnull;
+import org.axonframework.commandhandling.CommandHandler;
+import org.axonframework.commandhandling.CommandHandlingComponent;
+import org.axonframework.commandhandling.CommandMessage;
+import org.axonframework.commandhandling.CommandResultMessage;
+import org.axonframework.commandhandling.SimpleCommandHandlingComponent;
+import org.axonframework.eventhandling.EventHandler;
+import org.axonframework.eventhandling.EventHandlingComponent;
+import org.axonframework.eventhandling.EventMessage;
+import org.axonframework.eventhandling.SimpleEventHandlingComponent;
+import org.axonframework.messaging.MessageStream;
+import org.axonframework.messaging.QualifiedName;
+import org.axonframework.messaging.unitofwork.ProcessingContext;
+
+import java.util.Set;
+
+/**
+ * TODO This should be regarded as a playground object to verify the API.
+ *
+ * @author Steven van Beelen
+ * @since 5.0.0
+ */
+public class CommandModelComponent implements CommandHandlingComponent, EventHandlingComponent {
```

Comment:
The composition looks amazing, pretty clear that CommandModel is command + event handling. :)

Reply:
Thanks! Kinda pleased with it myself too. We're definitely not there yet though, of course!